### PR TITLE
Test/py3

### PR DESCRIPTION
**Pull request by mr-c**
Raw table access has been disabled for now

### CODE REVIEW

**Comment by mr-c** [comment]
For the new PrePartitionInfo CPython type:

- [ ] the CPython object name is of the form `khmer_${OBJECTNAME}_Object`
- [ ] Named struct with `PyObject_HEAD` macro
- [ ] `static PyTypeObject khmer_${OBJECTNAME}_Type` with the following
  entries
   - [ ] `PyVarObject_HEAD_INIT(NULL, 0)` as the object init (this includes
     the `ob_size` field).
   - [ ] all fields should have their name in a comment for readability
   - [ ] The `tp_name` filed is a dotted name with both the module name and
     the name of the type within the module. Example: `khmer.ReadAligner`
   - [ ] Deallocator defined and cast to `(destructor)` in tp_dealloc
     - [ ] The object's deallocator must be
       `Py_TYPE(obj)->tp_free((PyObject*)obj);`
   - [ ] Do _not_ define a `tp_getattr`
   - [ ] BONUS: write methods to present the state of the object via
     `tp_str` & `tp_repr`
   - [ ] _Do_ pass in the array of methods in `tp_methods`
   - [ ] _Do_ define a new method in `tp_new`
- [ ] PyMethodDef arrays contain doc strings
   - [ ] Methods are cast to `PyCFunctions`s
- [ ] Type methods use their type Object in the method signature.
- [ ] Type creation method decrements the reference to self
  (`Py_DECREF(self);`) before each error-path exit (`return NULL;`)
- [ ] No factory methods. Example: `khmer_new_readaligner`
- [ ] Type object is passed to `PyType_Ready` and its return code is checked
  in `init_khmer()`
- [ ] The reference count for the type object is incremented before adding
  it to the module: `Py_INCREF(&khmer_${OBJECTNAME}_Type);`.

**Comment by mr-c** [comment]
the new PrePartitionInfo CPython type is in the spirit of the KSubSetPartition CPython type: just a handle to the underlying C++ object. Not creatable in Python land.

- [ ] Update CPython Type checklist for types that are just handles for C++ pointers.

**Comment by luizirber** [comment]
Current status:
  - failing tests are annotated with 'failing_python2' and 'failing_python3' attributes.
  - mostly they involve unicode/bytes issues and division.

**Comment by mr-c** [comment]
@luizirber Can you summarize the status of the py3 compatibility? Checkboxes for the remaining tests would be super.

I also remember you saying that there were issues found in Screed via the khmer tests and that we need to port some tests back to screed for future

**Comment by luizirber** [comment]
See #978 for updated discussion (and squashed branch)
